### PR TITLE
User can view their profile information & statistics and update their information through settings

### PR DESCRIPTION
**Pull request by nadintamer**
## Description
This PR adds an initial interface for the following user stories:
* User can switch between profile view and settings view through a drawer navigation view
* User can view statistics about the number of words they've learned on the profile screen
* User can update their target language, e-mail, and password on the settings screen
* User can log out from the drawer navigation view

**Additional notes:**
* Error checking is implemented for updating user information in settings
* I ran into a bug when trying to implement changing the password and found out that changing the password logs the user out of the current session, which leads to a bunch of errors when trying to fetch data ([link to StackOverflow](https://stackoverflow.com/questions/32554305/invalid-session-token-in-parse-after-changing-password)). The fix was to log the user in again in the background with their new password.

## Testing
* Verified that changes don't break previous features
* Verified that updates to user information are reflected in the Parse database on Back4App

## Walkthrough GIF
profile-pr

### CODE REVIEW

File: app/src/main/java/com/example/lexis/fragments/ProfileSettingsFragment.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+import org.parceler.Parcels;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class ProfileSettingsFragment extends Fragment {
+
+    private static final String ARG_USER = "user";
+    private static final List<String> languageCodes = Arrays.asList("fr", "es", "de", "tr");
```
Comment (annechen112995) [inline app/src/main/java/com/example/lexis/fragments/ProfileSettingsFragment.java:36]: I think this also shows up in the signup activity: would it be useful to pull this out to a constant file somewhere?

Comment (annechen112995) [reply app/src/main/java/com/example/lexis/fragments/ProfileSettingsFragment.java:36]: That way, if you want to change the languages you support, you don't have to change multiple callsites.

**Review by annechen112995 — APPROVED** [review]
image

File: app/src/main/java/com/example/lexis/fragments/ProfileSettingsFragment.java (hunk 1 above)
Comment (nadintamer) [reply app/src/main/java/com/example/lexis/fragments/ProfileSettingsFragment.java]: Yes, that makes sense — fixed!

**Pull request merged**